src/main/java/io/antmedia/streamsource/StreamFetcher.java

```diff
@@ -80,20 +55,43 @@
 	private long[] lastDTS;
 	private long[] lastPTS;
 	private MuxAdaptor muxAdaptor = null;
-	
+
 	/**
 	 * If it is true, it restarts fetching everytime it disconnects
 	 * if it is false, it does not restart
 	 */
 	private boolean restartStream = true;
 
-	public StreamFetcher(Broadcast stream, IScope scope) throws Exception {
+	/**
+	 * Buffer time in milliseconds
+	 */
+	private int bufferTime = 0;
+	
+	private ConcurrentLinkedQueue<AVPacket> availableBufferQueue = new ConcurrentLinkedQueue<>();
+
+	private ISchedulingService scheduler;
+	private AVRational avRationalTimeBaseMS;
+	private AppSettings appSettings;
+
+	public StreamFetcher(Broadcast stream, IScope scope, ISchedulingService scheduler) throws Exception {
 		if (stream == null || stream.getStreamId() == null || stream.getStreamUrl() == null) {
 			throw new Exception("Stream is not initialized properly. Check stream("+stream+"), "
```

Comment:
An exception specific to Antmedia could be created by extending Exception class. In the future it will be easy to identify Antmedia specific exceptions.

Reply:
Fixed https://github.com/ant-media/Ant-Media-Server/blob/dev/src/main/java/io/antmedia/streamsource/StreamFetcher.java